### PR TITLE
Add support for Clear-Site-Data header

### LINKED ISSUE #284: Add support for Clear Site Data header
## Adding a new header

Clear-Site-Data: https://www.w3.org/TR/clear-site-data/
- Is the header supported by any user agent? If so, which?
  - [Maybe chrome](https://www.chromestatus.com/features/4713262029471744), definitely not the others
- What does it do?
  - "A ‘Clear-Site-Data’ HTTP header prompts the user agent to clear browsing data associated with the requesting website. The supported browsing data types are cookies, storage (i.e. “site data”), and cache. This is a privacy and security enhancing feature. A sensitive website can trigger local data deletion after the user signs out. A website dealing with a persistent XSS attack can use this to ‘reset’ itself to a clean state."
- What are the valid values for the header?
  - Looks like it accepts json: `{ "types": [ "cache", "cookies", "storage", "executionContexts" ] }`
- Where does the specification live?
  - https://www.w3.org/TR/clear-site-data/

### PR DESCRIPTION
**Pull request by mastahyeti**
fixes #284

## All PRs:

* [x] Has tests
* [x] Documentation updated

## Adding a new header

* Is the header supported by any user agent? If so, which?
  * As far as I know, this is a [preview-feature in Chrome](https://groups.google.com/a/chromium.org/forum/#!msg/blink-dev/NHkG8FXFEWU/fz9uDwBTBQAJ). The [Chrome Platform Status page](https://www.chromestatus.com/feature/4713262029471744) might need updating...
* What does it do?
  * Allows sites to clear cookies, cache, etc.. from the browser.
* What are the valid values for the header?
  * A JSON object with a `types` member that is an array of string types of site-data to clear. As of the current spec, the valid types are `cache`, `cookies`, `storage`, and `executionContexts`.
* Where does the specification live?
  * https://www.w3.org/TR/clear-site-data/

### CODE REVIEW

**Comment by reedloden** [comment]
This fixes #284

File: lib/secure_headers/headers/clear_site_data.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    EXECTION_CONTEXTS = "executionContexts".freeze
+    ALL_TYPES = [CACHE, COOKIES, STORAGE, EXECTION_CONTEXTS]
+
+    CONFIG_KEY = :clear_site_data
+
+    class << self
+      # Public: make an Clear-Site-Data header name, value pair
+      #
+      # Returns nil if not configured, returns header name and value if configured.
+      def make_header(config=nil)
```
Comment (oreoshake) [inline lib/secure_headers/headers/clear_site_data.rb]: :+1:

**Comment by oreoshake** [comment]
I'm not entirely sure how to push to this PR, I should probably figure that out, but this needs the following diff to be complete. It's mostly boilerplate 😢 

```diff
diff --git a/README.md b/README.md
index ad11584..5109101 100644
--- a/README.md
+++ b/README.md
@@ -69,6 +69,12 @@ SecureHeaders::Configuration.default do |config|
   config.x_download_options = "noopen"
   config.x_permitted_cross_domain_policies = "none"
   config.referrer_policy = "origin-when-cross-origin"
+  config.clear_site_data = [
+    "cache",
+    "cookies",
+    "storage",
+    "executionContexts"
+  ]
   config.csp = {
     # "meta" values. these will shaped the header, but the values are not included in the header.
     report_only: true,      # default: false [DEPRECATED from 3.5.0: instead, configure csp_report_only]
diff --git a/lib/secure_headers.rb b/lib/secure_headers.rb
index c51c1d0..84502f0 100644
--- a/lib/secure_headers.rb
+++ b/lib/secure_headers.rb
@@ -51,6 +51,7 @@ module SecureHeaders
   CSP = ContentSecurityPolicy
 
   ALL_HEADER_CLASSES = [
+    ClearSiteData,
     ContentSecurityPolicyConfig,
     ContentSecurityPolicyReportOnlyConfig,
     StrictTransportSecurity,
diff --git a/lib/secure_headers/configuration.rb b/lib/secure_headers/configuration.rb
index b0a53df..13447d8 100644
--- a/lib/secure_headers/configuration.rb
+++ b/lib/secure_headers/configuration.rb
@@ -116,7 +116,7 @@ module SecureHeaders
 
     attr_writer :hsts, :x_frame_options, :x_content_type_options,
       :x_xss_protection, :x_download_options, :x_permitted_cross_domain_policies,
-      :referrer_policy
+      :referrer_policy, :clear_site_data
 
     attr_reader :cached_headers, :csp, :cookies, :csp_report_only, :hpkp, :hpkp_report_host
 
@@ -150,6 +150,7 @@ module SecureHeaders
       copy.x_xss_protection = @x_xss_protection
       copy.x_download_options = @x_download_options
       copy.x_permitted_cross_domain_policies = @x_permitted_cross_domain_policies
+      copy.clear_site_data = @clear_site_data
       copy.referrer_policy = @referrer_policy
       copy.hpkp = @hpkp
       copy.hpkp_report_host = @hpkp_report_host
@@ -181,6 +182,7 @@ module SecureHeaders
       XXssProtection.validate_config!(@x_xss_protection)
       XDownloadOptions.validate_config!(@x_download_options)
       XPermittedCrossDomainPolicies.validate_config!(@x_permitted_cross_domain_policies)
+      ClearSiteData.validate_config!(@clear_site_data)
       PublicKeyPins.validate_config!(@hpkp)
       Cookie.validate_config!(@cookies)
     end
diff --git a/spec/lib/secure_headers_spec.rb b/spec/lib/secure_headers_spec.rb
index 5715d83..c6a04a2 100644
--- a/spec/lib/secure_headers_spec.rb
+++ b/spec/lib/secure_headers_spec.rb
@@ -530,6 +530,14 @@ module SecureHeaders
         end.to raise_error(XContentTypeOptionsConfigError)
       end
 
+      it "validates your clear site data config upon configuration" do
+        expect do
+          Configuration.default do |config|
+            config.clear_site_data = 1
+          end
+        end.to raise_error(ClearSiteDataConfigError)
+      end
+
       it "validates your x_xss config upon configuration" do
         expect do
           Configuration.default do |config|
```

**Comment by oreoshake** [comment]
3.6.0 has been released with this feature

**Pull request merged**